pyvista/plotting/tools.py

```diff
@@ -10,29 +11,71 @@
 from .theme import parse_color, rcParams
 
 
-def system_supports_plotting():
-    """Check if x server is running.
+# Track render window support and plotting
+SUPPORTS_OPENGL = [None]
+SUPPORTS_PLOTTING = [None]
+
+def supports_open_gl():
+    """Return if the system supports OpenGL."""
+    if SUPPORTS_OPENGL[0] is None:
+        ren_win = _vtk.vtkRenderWindow()
+        SUPPORTS_OPENGL[0] = bool(ren_win.SupportsOpenGL())
+    return SUPPORTS_OPENGL[0]
+
+
+def _system_supports_plotting():
+    """Check if the environment supports plotting on Windows, Linux, or Mac OS.
 
     Returns
     -------
     system_supports_plotting : bool
-        True when on Linux and running an xserver.  Returns None when
-        on a non-linux platform.
+        ``True`` when system supports plotting.
 
     """
-    try:
-        if os.environ['ALLOW_PLOTTING'].lower() == 'true':
+    if os.environ.get('ALLOW_PLOTTING', '').lower() == 'true':
+        return True
+
+    # Windows case
+    if os.name == 'nt':
+        # actually have to check here.  Somewhat expensive.
+        return supports_open_gl()
+
+    # mac case
+    if platform.system() == 'Darwin':
```

Comment:
Following suggestions and patterns here:
https://stackoverflow.com/a/1879/3369879

It's true that we could just use platform, but I'd prefer using os and platform separately since `os.name` is consistent for linux and nt, and `platform.system` returns a single result for Mac OS.  Effectively, I'm trying to avoid https://stackoverflow.com/a/25863224/3369879.